src/main/java/spoon/support/reflect/declaration/CtTypeImpl.java

```diff
@@ -894,6 +895,14 @@ public String getQualifiedName() {
 		map(new AllTypeMembersFunction(CtMethod.class)).forEach(new CtConsumer<CtMethod<?>>() {
 			@Override
 			public void accept(CtMethod<?> method) {
+				ClassTypingContext ctc = new ClassTypingContext(CtTypeImpl.this);
+
```

Comment:
I suggest to move creation of ClassTypingContext above (out of the cycle). It will perform faster, because creation of ClassTypingContext is expensive and it is better to use cached information about super type hierarchy, then to create it always again. 

Like this:
```java
	@Override
	public Set<CtMethod<?>> getAllMethods() {
		final Set<String> distinctSignatures = new HashSet<>();
		final Set<CtMethod<?>> l = new SignatureBasedSortedSet<>();
		final ClassTypingContext ctc = new ClassTypingContext(CtTypeImpl.this);
		map(new AllTypeMembersFunction(CtMethod.class)).forEach(new CtConsumer<CtMethod<?>>() {
			@Override
			public void accept(CtMethod<?> method) {

				for (CtMethod<?> methods : l) {
					if (ctc.isSameSignature(methods, method)) {
						return;
					}
				}

				if (distinctSignatures.add(method.getSignature())) {
					l.add(method);
				}
			}
		});
		return Collections.unmodifiableSet(l);
	}
```

I would commit into this branch, but I have no permission.